go/obscuronode/enclave/enclave.go

```diff
@@ -919,9 +919,18 @@ func getDB(nodeID uint64, cfg config.EnclaveConfig) (ethdb.Database, error) {
 	}
 
 	// persistent and with attestation means connecting to edgeless DB in a trusted enclave from a secure enclave
-	panic("Haven't implemented edgeless DB enclave connection yet")
+	nodecommon.LogWithID(nodeID, "Preparing Edgeless DB connection...")
+	return getEdgelessDB(cfg)
 }
 
 func getInMemDB() (ethdb.Database, error) {
 	return rawdb.NewMemoryDatabase(), nil
 }
+
+func getEdgelessDB(cfg config.EnclaveConfig) (ethdb.Database, error) {
+	if cfg.EdgelessDBHost == "" {
+		return nil, fmt.Errorf("failed to prepare EdgelessDB connection - EdgelessDBHost was not set on enclave config")
+	}
+	dbConfig := sql.EdgelessDBConfig{Host: cfg.EdgelessDBHost}
+	return sql.EdgelessDBConnector(dbConfig)
```

Comment:
@BedrockSquirrel Shouldn't we be attempting to clear up the datadir if there was an error of any kind?

Reply:
I don't think so, whatever set up the ego enclave had to mount the data dir onto it to make it available, I think it's fair for them to manage its lifecycle since it's usually docker and it can be helpful to poke through it afterwards when debugging.

You could definitely argue that we don't need to attempt to create the dir and we should assume it's there... But don't think it does much harm in trying to create it for now.